Merge

NathanHinthorne opened the pull request:
Update slingshot to fire projectiles. Added vector class. Added inventory class.

b1gd3vd0g committed:
Merge pull request #5 from GoodBadChad/main

Sync

Commit:
Add sprinting mechanic.

Commit:
Update to dash: bug fix

Dash should only be usable in one direction while in the air. Bug was allowing player to dash in multiple directions as long as the player was not on the ground and hasn't hit the dash limit barrier. Bug fixed.

Commit:
Update docs and naming.

b1gd3vd0g committed:
Merge pull request #60 from calebkrauter/development

Implement Sprint and Dash mechanics. Debugged.

tclaar committed:
Added Vector class.

tclaar committed:
Merge pull request #61 from tclaar/main

Push Vector class.

b1gd3vd0g committed:
Merge pull request #62 from GoodBadChad/trae

Merge Vector class.

b1gd3vd0g committed:
Merge pull request #6 from GoodBadChad/merge

Merge

b1gd3vd0g committed:
Blacksmith house (this is atually trae's work)

b1gd3vd0g committed:
bs house sprite

b1gd3vd0g committed:
load village first

b1gd3vd0g committed:
loadvillagedimension script

b1gd3vd0g committed:
Merge branch 'main' of https://github.com/B1GD3VD0G/good-bad-chad-devin

tclaar committed:
Refactored codebase to use Vector class.
Refactored all scripts (except dimension.js) to utilize the Vector class.
Added round() static method to Vector class.

tclaar committed:
Merge pull request #63 from tclaar/main

Refactored codebase to use Vector class.

tclaar committed:
Merge branch 'merge' into trae

b1gd3vd0g committed:
Merge branch 'main' into trae

Commit:
Debug Game.

The PR to merge contained issues where some code still needed to be refactored in PapaChad.js to use the vector class functionality. The game was loading Village which is not in the same adequate playable state as PlayGround. I updated the code in PapaChad to use "pos" where necessaary with the new Vector class. The dash functionality was broken so I removed code modularity due to a scope issue with xVelocity variable and put the functionality of DashAction back into the update loop code and removed a buggy portion of DashAction. I commented out some console.log code in slinshot that used undefined variables.

calebkrauter committed:
Merge pull request #64 from GoodBadChad/trae

Merge Vector class and associated refactoring.

calebkrauter commented on js/classes/animator.js:
```diff
@@ -8,24 +8,18 @@ class Animator {
-     * @param {number} yStart The starting y position (on the SPRITESHEET!) of the first sprite of the animation.
-     * @param {number} width The width in pixels of the sprites in this animation.
-     * @param {number} height The height in pixels of the sprites in this animation.
+     * @param {Vector} start The starting position (on the SPRITESHEET!) of the first sprite of the animation.
```
Is Vector a type? I'm a bit unfamiliar with this format of {type}. Or is it more flexible including any time of Object etc? @B1GD3VD0G

calebkrauter commented on js/classes/animator.js:
```diff
@@ -8,24 +8,18 @@ class Animator {
+        /** The starting position (on the SPRITESHEET!) of the first sprite of the animation. */
+        this.start = start;
+        /** The size in pixels of the sprites in this animation. */
+        this.size = size;
```
I like this! Much cleaner using size then height and width.

calebkrauter commented on js/classes/animator.js:
```diff
@@ -38,19 +32,18 @@ class Animator {
      * This method is going to draw the appropriate frame on the Canvas (to which ctx belongs to).
-     * @param {number} x The x position (of the CANVAS!) at which we'd like our sprite to be drawn.
-     * @param {number} y The y position (of the CANVAS!) at which we'd like our sprite to be drawn.
+     * @param {Vector} x The position (of the CANVAS!) at which we'd like our sprite to be drawn.
```
Is this Vector x or "pos"

calebkrauter commented on js/classes/animator.js:
```diff
@@ -38,19 +32,18 @@ class Animator {
-    drawFrame(x, y, scale) {
+    drawFrame(pos, scale) {
         this.elapsedTime += GAME.clockTick;
         if (this.elapsedTime > this.totalTime) this.elapsedTime -= this.totalTime;
```
Is this restarting the duration between frames so that each frame time isn't too long?

calebkrauter commented on js/classes/animator.js:
```diff
@@ -38,19 +32,18 @@ class Animator {
-            this.width, this.height,
-            x, y,
-            this.width * scale, this.height * scale);
+            this.start.x + this.size.x * frame, this.start.y,
```
Wondering if there is any way to make this more readable, it's confusing to look all these parameters and try to understand what it is doing specifically. What is frame?

calebkrauter commented on js/classes/bounding_box.js:
```diff
@@ -5,24 +5,19 @@
+        this.top = pos.y;
         /** The y coordinate of the bottom boundary of the box. */
-        this.bottom = y + height;
+        this.bottom = pos.y + size.y;
```
This is great! Using the Vector class. Love it. @B1GD3VD0G

b1gd3vd0g commented on js/classes/entities/background/village/bird.js:
```diff
@@ -8,98 +8,88 @@ class Bird {
+    };
 
     /** The speed (in pixels/frame) of a Bird. */
     static get SPEED() {
```
I think that speed should be relative to scale; that way, when you change scale, the entity does not appear way too slow/fast.

b1gd3vd0g commented on js/classes/entities/background/village/bird.js:
```diff
@@ -8,98 +8,88 @@ class Bird {
      * 
      * @return {number} the actual width of the Bird when drawn
      */
     actualWidth() {
```
Static constant?

calebkrauter replied:
Good idea! @tclaar

calebkrauter commented on js/classes/camera.js:
```diff
@@ -38,7 +35,8 @@ class Camera {
             }
             return z;
```
Why are we returning at the end of the Camera class and what is z for? What does it mean or do? @B1GD3VD0G

b1gd3vd0g commented on js/classes/entities/background/village/bird.js:
```diff
@@ -8,98 +8,88 @@ class Bird {
      * @param {number} maxY the maximum y position (in pixels) a bird can achieve before resetting
      */
-    constructor(maxX, maxY) {
+    constructor(minX, minY, maxX, maxY) {
```
Could be vectors: minRange, maxRange, etc. Not super important.

tclaar commented on js/classes/entities/midground/npc/village/papa_chad.js:
```diff
@@ -88,42 +97,107 @@ class PapaChad {
         // Step 1: Listen for user input.
-        // for now, just left and right
-        this.xVelocity = 0;
+        if (this.isOnGround) {
```
It looks to me like the number of if statements and condition checks within this method (particularly for double-jump functionality, dash functionality, etc.) can be reduced somewhat. This should be looked into at a later point to increase readability and maintainability. Not pertinent for this merge however as it functions fine. @calebkrauter @B1GD3VD0G

b1gd3vd0g commented on js/classes/entities/background/village/houses.js:
```diff
@@ -0,0 +1,28 @@
+class BSHouse {
+    constructor(x, y) {
```
Should this be a Vector?

tclaar commented on js/classes/entities/background/village/houses.js:
```diff
@@ -0,0 +1,28 @@
+class BSHouse {
```
This class should be removed as it will be rendered obsolete with the addition of the Decoration class.

calebkrauter replied:
Missing Documentation.

b1gd3vd0g commented on js/classes/entities/foreground/crosshair.js:
```diff
@@ -4,17 +4,17 @@
     constructor() {
-        this.x = GAME.mouseX;
-        this.y = GAME.mouseY;
+        // this.x = GAME.mouseX;
```
these comments should be deleted.

b1gd3vd0g commented on js/classes/entities/foreground/dialog_bubble.js:
I think that these should, in general, be bigger. Personally I imagined them taking up the entire width of the screen (minus a reasonable margin) on the bottom. This way, the text could be written bigger, and more info can be written in a single bubble when necessary.  I don't think that dynamic sizing is necessary, and I honestly think it can be distracting or a little offputting.

calebkrauter replied:
Perhaps we can discuss it some more. I am torn on this.

tclaar commented on js/classes/camera.js:
The internals of this class need to be refactored to fully take advantage of the Vector class. This should be done when it is modified to work with Zones. @B1GD3VD0G

b1gd3vd0g commented on js/classes/entities/midground/enemies/slime.js:
```diff
@@ -80,25 +77,30 @@ class Slime {
+    // /** This will be the width of the Slime ON THE CANVAS. */
+    // static get SCALED_WIDTH() {
+    //     return Slime.SCALE * Slime.WIDTH;
+    // };
```
Need to delete these comments

tclaar commented on js/classes/entities/foreground/crosshair.js:
```diff
@@ -44,12 +44,15 @@ class Crosshair {
-    static get WIDTH() {
-        return 16;
-    }
+    // static get WIDTH() {
```
Commented-out code should be removed. My bad...

b1gd3vd0g commented on js/classes/entities/midground/enemies/village/bunny.js:
```diff
@@ -82,13 +67,13 @@ class Bunny {
     };
 
     /** Draw the Bunny on the canvas. */
     draw() {
```
Bunny was made for the animation assignment, and does not draw based on the position of the camera yet.

tclaar commented on js/classes/entities/midground/enemies/slime.js:
```diff
@@ -80,25 +77,30 @@ class Slime {
-    static get SCALED_HEIGHT() {
-        return Slime.SCALE * Slime.HEIGHT;
-    };
+    // /** This will be the height of the Slime ON THE CANVAS. */
```
Commented-out code should be removed.

calebkrauter commented on js/classes/entities/background/village/bird.js:
```diff
@@ -8,98 +8,88 @@ class Bird {
-            this.y = this.randomY();
-            targetX = -this.actualWidth();
+        if (Math.random() > 0.5) {
+            // or on the right
```
I am confused by this comment. Was there supposed to be a left side in a different conditional that has been removed? Why does Math.random() have to be less than or greater than 0.5 specifically? @tclaar

b1gd3vd0g commented on js/classes/entities/midground/enemies/village/snake.js:
```diff
@@ -82,13 +68,13 @@ class Snake {
     };
 
     /** Draw the Snake on the canvas. */
     draw() {
```
Snake was made for the animation assignment, and does not draw based on CAMERA.pos

calebkrauter commented on js/classes/entities/background/village/bird.js:
```diff
@@ -8,98 +8,88 @@ class Bird {
     /**
      * Constructor for a Bird.
      * 
+     * @param {number} minX the minimum x position (in pixels) a bird can achieve before resetting
```
It seems like this is the minimum x value that the bird can travel to before resetting implying that birds have a minimum and maximum distance? This mostly makes sense, could be a little more clear perhaps. @tclaar

tclaar commented on js/classes/entities/midground/enemies/village/bunny.js:
```diff
@@ -82,13 +67,13 @@ class Bunny {
     /** Draw the Bunny on the canvas. */
     draw() {
-        this.animations[this.facing][this.action].drawFrame(this.x, this.y, Bunny.SCALE);
+        this.animations[this.facing][this.action].drawFrame(this.pos, Bunny.SCALE);
```
The Bunny's position should be converted to canvas space before passing to the `drawFrame` method, e.g.: 
`this.animations[this.facing][this.action].drawFrame(Vector.worldToCanvasSpace(this.pos), Bunny.SCALE);`

calebkrauter commented on js/classes/entities/background/village/bird.js:
```diff
@@ -8,98 +8,88 @@ class Bird {
         this.maxX = maxX;
         this.maxY = maxY;
-        this.animator = new Animator(Bird.SPRITESHEET, 0, 0, Bird.WIDTH, Bird.HEIGHT, 8, 1/16);
+        this.animator = new Animator(Bird.SPRITESHEET, 0, 0, Bird.SIZE.x, Bird.SIZE.y, 8, 1/16);
```
Perhaps we could start using constants for some of these or all of these values in all of our classes that use "animator" for better readability. @tclaar

tclaar commented on js/classes/entities/midground/enemies/village/snake.js:
```diff
@@ -82,13 +68,13 @@ class Snake {
     /** Draw the Snake on the canvas. */
     draw() {
-        this.animations[this.facing][this.action].drawFrame(this.x, this.y, Snake.SCALE);
+        this.animations[this.facing][this.action].drawFrame(this.pos, Snake.SCALE);
```
The Snake's position should be converted to canvas space before passing to the `drawFrame` method, e.g.: 
`this.animations[this.facing][this.action].drawFrame(Vector.worldToCanvasSpace(this.pos), Snake.SCALE);`

calebkrauter replied:
Good point!

b1gd3vd0g commented on js/classes/entities/midground/npc/village/papa_chad.js:
```diff
@@ -79,6 +74,20 @@ class PapaChad {
+    }
+
     /** Change what Papa Chad is doing and where it is. */
     update() {
```
THIS update method (meaning the one listening to user controls, controlling main character - not necessarily PapaChad's) is inherently extremely complex. It should be more thouroughly commented. It seems like code has been stuck here after comments saying it's doing one thing, but really it's doing another. I think that it should be cleaned up and commented through thouroughly upon its transfer to Chad's class.

b1gd3vd0g replied:
There are a lot of loose variables here. Maybe try cleaning that up a bit - instead of canJump, canDoubleJump, etc., try maybe this.permissions being an object with booleans jump, dash, etc.

calebkrauter commented on js/classes/entities/background/village/bird.js:
```diff
@@ -8,98 +8,88 @@ class Bird {
     update() {
-        this.x += this.dirX * Bird.SPEED;
-        this.y += this.dirY * Bird.SPEED;
+        this.pos = Vector.add(this.pos, Vector.multiply(this.dir, Bird.SPEED));
```
I am torn on this. I like it because it is concise and simple however should we clarify it with constants or perhaps with a comment? @tclaar

tclaar commented on js/classes/game_engine.js:
Some Vector refactoring still needs to be done in here, specifically related to the old Dimension functionality. This refactoring should be done alongside modification to work with Zones. @B1GD3VD0G

calebkrauter commented on js/classes/entities/background/village/bird.js:
```diff
@@ -8,98 +8,88 @@ class Bird {
     randomY() {
-        return Math.random() * this.maxY;
-    }
+        return this.minY + Math.random() * (this.maxY - this.minY);
```
I like this. Good idea. @tclaar

tclaar reviewed (approved):
Looks ready to merge. Some changes need to be made in the future (mostly just refactoring, particularly in camera.js, game_engine.js, and papa_chad.js) but can wait for now.

b1gd3vd0g commented on js/classes/entities/midground/npc/village/papa_chad.js:
```diff
@@ -66,10 +65,6 @@ class PapaChad {
-    static get WIDTH() {
-        return 29;
-    };
     /** The velocity of the first jump */
```
All of these getters - FIRST_JUMP_VELOCITY thru DASH_BARRIER - ought be static. Their return does not depend upon the state of this {Papa}Chad instance - therefore, they should be static and called by saying Chad.FIRST_JUMP_VELOCITY, not CHAD.FIRST_JUMP_VELOCITY

calebkrauter commented on js/classes/entities/background/village/houses.js:
```diff
@@ -0,0 +1,28 @@
+    constructor(x, y) {
+        this.x = x * Block.SCALED_SIZE;
+        this.y = y * Block.SCALED_SIZE;
+        this.animator = new Animator(BSHouse.SPRITESHEET, 0, 0, BSHouse.SIZE, BSHouse.SIZE,
```
Could this be updated to use constants when making an instance of Animator? @B1GD3VD0G

calebkrauter commented on js/classes/entities/background/village/houses.js:
```diff
@@ -0,0 +1,28 @@
+        return 3;
+    };
+
+    update() {
```
Do we need an update method if it isn't being used? @B1GD3VD0G

calebkrauter commented on js/classes/entities/background/village/houses.js:
```diff
@@ -0,0 +1,28 @@
+
+    };
+
+    draw() {
```
Could we refactor this with the vector class? Perhaps I am missing something. @B1GD3VD0G

b1gd3vd0g commented on js/classes/game_engine.js:
```diff
@@ -20,14 +20,19 @@ class GameEngine {
         this.right = false;
```
THESE USER CONTROLS ARE NAMED BADLY. Each one should dictate EITHER (a) what the user is doing to their computer, or (b) what the user is trying to make CHAD do. I personally vote for (b). @NathanHinthorne @calebkrauter @tclaar

b1gd3vd0g replied:
Also to clean up namespace, consider using a single object like User/ UserInput = {jumping: x, movingLeft: x, movingRight: x, ...}

calebkrauter replied:
Good ideas @B1GD3VD0G . Initially I preferred that variables represent what the user of the game/program is doing. After more thought and discussing it with you, making variables named after behavior in game makes sense especially if we decide to allow key-remapping.

b1gd3vd0g commented on js/scripts/tileMaps/playground.js:
We should change this folder to tile_maps instead of tileMaps.

calebkrauter replied:
Thanks! Good catch.

b1gd3vd0g reviewed (approved):
Good. I've left comments, but it's good for approval to main as it is bug free.

calebkrauter committed:
Delete js/classes/entities/background/village/houses.js

Talked with @tclaar. He requested we delete this file so I did it during my PR.

NathanHinthorne committed:
Merge pull request #75 from GoodBadChad/calebkrauter-patch-2

Delete js/classes/entities/background/village/houses.js

calebkrauter commented on js/classes/entities/foreground/dialog_bubble.js:
```diff
@@ -27,12 +27,10 @@ class DialogBubble {
-        this.y = CTX.canvas.height - (DialogBubble.HEIGHT - 50) * this.scale; // bottom
+        
+        /** The position of the sprite ON THE SPRITESHEET. */
+        this.pos = new Vector(CTX.canvas.width / 2 - DialogBubble.SIZE.x * this.scale / 2,
```
Perhaps we could refactor this or do something to improve readability like name some variables for the calculations to make it more clear what is happening?

calebkrauter commented on js/classes/entities/foreground/dialog_bubble.js:
```diff
@@ -77,28 +75,28 @@ class DialogBubble {
-            this.x, this.y - (DialogBubble.HEIGHT * 3) - 60,
-            DialogBubble.WIDTH * this.scale, DialogBubble.HEIGHT*3);
+            DialogBubble.SIZE.x, DialogBubble.SIZE.y,
+            this.pos.x, this.pos.y - (DialogBubble.SIZE.y * 3) - 60,
```
Would be good to consider removing "magic numbers" that are not assigned a clear variable.

calebkrauter commented on js/classes/entities/foreground/dialog_bubble.js:
```diff
@@ -107,16 +105,16 @@ class DialogBubble {
         // read in the text and draw it line by line
         let lines = this.text.split("\n");
```
Good idea! I like it. @NathanHinthorne

calebkrauter commented on js/classes/entities/midground/enemies/slime.js:
```diff
@@ -80,25 +77,30 @@ class Slime {
+    }
 
     static get SPEED() {
         return Slime.SCALE * 30;
```
Why is a value multiplied by Slime.SCALE to produce the speed? Does this work well? Just curious.

calebkrauter commented on js/classes/entities/midground/enemies/slime.js:
```diff
@@ -133,13 +130,13 @@ class Slime {
         // HOW SHOULD WE MOVE HIM BASED ON HIS CONDITIONS?
         const mult = (this.facing === "left") ? -1 : 1;
         if (this.action === "moving") {
```
What is the difference between == and === ? Is === more specific?

calebkrauter commented on js/classes/entities/midground/enemies/slime.js:
```diff
@@ -133,13 +130,13 @@ class Slime {
         // HOW SHOULD WE MOVE HIM BASED ON HIS CONDITIONS?
         const mult = (this.facing === "left") ? -1 : 1;
```
I would suggest changing the name "mult". Something like direction possibly. Unless I am misunderstanding this code.

calebkrauter commented on js/classes/entities/midground/enemies/slime.js:
```diff
@@ -133,13 +130,13 @@ class Slime {
         const mult = (this.facing === "left") ? -1 : 1;
         if (this.action === "moving") {
-            this.x += mult * Slime.SPEED * GAME.clockTick;
+            this.pos = Vector.add(this.pos, new Vector(mult * Slime.SPEED * GAME.clockTick, 0));
```
I am a bit confused by this line. Why are we multiplying SPEED by GAME.clockTick? It appears that this line of code is causing the slime to move either left or right at some speed. Do we need GAME.clockTick for the speed if we are in the update loop?

calebkrauter commented on js/classes/entities/midground/enemies/slime.js:
```diff
@@ -150,25 +147,25 @@ class Slime {
             Slime.SPRITESHEET,
-            0, this.type * 2 * Slime.HEIGHT + Slime.HEIGHT,
-            Slime.WIDTH, Slime.HEIGHT,
+            new Vector(0, this.type * 2 * Slime.SIZE.y + Slime.SIZE.y),
```
We could clean up some of the "magic numbers" that have no variables potentially in the parameters of the instantiation of Animator.

calebkrauter commented on js/classes/entities/midground/enemies/village/bunny.js:
```diff
@@ -82,13 +67,13 @@ class Bunny {
         const mult = (this.facing === "left") ? -1 : 1;
         if (this.action === "hopping") {
-            this.x += mult * Bunny.SPEED * GAME.clockTick;
+            this.pos = Vector.add(this.pos, new Vector(mult * Bunny.SPEED * GAME.clockTick, 0));
```
Same question about the necessity of clockTick like in slime.js

calebkrauter commented on js/classes/entities/midground/enemies/village/bunny.js:
```diff
@@ -98,26 +83,26 @@ class Bunny {
-            Bunny.WIDTH, Bunny.HEIGHT,
+            Bunny.SIZE,
+            Bunny.SIZE,
             4, 0.125);
```
It just hit me that we will want to consider if all of our animations should have the same amount of frames and frame times. Maybe all of them won't but we may want to have consistency for different types of animations or different creatures. CHAD will likely be the smoothest or one of the smoothest. @tclaar @NathanHinthorne @B1GD3VD0G

calebkrauter commented on js/classes/entities/midground/portal.js:
```diff
@@ -1,16 +1,15 @@
 class Portal {
```
This class appears to only draw the portal and not  have anything to do with the functionality of it. Might be worth considering a rename to the class.

calebkrauter commented on js/classes/entities/midground/weapon/projectile.js:
```diff
@@ -26,46 +24,47 @@ class Projectile {
+    };
 
     /** The Laser Projectile type. */
     static get LASER() {
```
Han Shot first...

calebkrauter commented on js/classes/entities/midground/weapon/projectile.js:
```diff
@@ -26,46 +24,47 @@ class Projectile {
+    };
 
     /** The property table for Projectile types. */
     static get PROPERTY_TABLE() {
```
Love this. I'm wanna start using this type of mapping.

calebkrauter commented on js/classes/entities/midground/weapon/projectile.js:
```diff
@@ -165,8 +159,8 @@ class Projectile {
+            this.pos = Vector.add(this.pos, new Vector(posChange.x, posChange.y + this.yVelocity))
 
             this.updateBoundingBox();
             GAME.entities.forEach((entity) => {
```
Is there any specific reason this is done in the form of an arrow function? @tclaar Just curious.

calebkrauter commented on js/classes/entities/midground/weapon/projectile.js:
```diff
@@ -76,56 +75,51 @@ class Projectile {
+                SIZE: new Vector(13, 13)
             },
             [Projectile.WOOD]: {
                 ACTION: () => {
```
Is ACTION just a keyword to a function? Like a key to the arrow function? Very interesting. I wonder what other uses this could have. @tclaar

calebkrauter commented on js/classes/entities/midground/weapon/slingshot.js:
```diff
@@ -6,53 +6,53 @@ class Slingshot {
+        const delta = Vector.worldToCanvasSpace(Vector.subtract(GAME.mousePos, this.pos));
+        // There is an undefined value here so I commented this out - CK
+        // console.log(GAME.mouse);
+        let theta = Math.atan2(delta.y, delta.x);
```
Smart! Nice work @NathanHinthorne

calebkrauter commented on js/classes/entities/midground/weapon/slingshot.js:
```diff
@@ -6,53 +6,53 @@ class Slingshot {
-        let start = { x: Math.round(this.x), y: Math.round(this.y) };
-        let end = { x: Math.round(GAME.mouseX + CAMERA.x), y: Math.round(GAME.mouseY + CAMERA.y) };
-        GAME.addEntity(new Projectile(Projectile.STONE, start.x, start.y, end.x, end.y));
+        GAME.addEntity(new Projectile(
```
I like this. Lots of words to read but pretty clear what is happening. Very cool!

calebkrauter commented on js/classes/game_engine.js:
```diff
@@ -102,7 +107,7 @@ class GameEngine {
         CTX.clearRect(0, 0, CANVAS.width, CANVAS.height);
         CTX.fillStyle = "#00ff00";
-        CTX.fillRect(DIMENSION.MIN_X - CAMERA.x, DIMENSION.MIN_Y - CAMERA.y, DIMENSION.BLOCK_WIDTH * Block.SCALED_SIZE, DIMENSION.BLOCK_HEIGHT * Block.SCALED_SIZE);
+        CTX.fillRect(DIMENSION.MIN_X - CAMERA.pos.x, DIMENSION.MIN_Y - CAMERA.pos.y, DIMENSION.BLOCK_WIDTH * Block.SCALED_SIZE, DIMENSION.BLOCK_HEIGHT * Block.SCALED_SIZE);
```
What is this? Just the camera view?

calebkrauter commented on js/classes/game_engine.js:
```diff
@@ -188,13 +206,13 @@ class GameEngine {
         for (let x = DIMENSION.MIN_X; x <= DIMENSION.MAX_X; x += Block.SCALED_SIZE) {
```
Is this for the debugging @B1GD3VD0G ?

calebkrauter commented on js/classes/game_engine.js:
```diff
@@ -204,14 +222,14 @@ class GameEngine {
+            for (let blockY = DIMENSION.MIN_BLOCK_Y; blockY <= DIMENSION.MAX_BLOCK_Y; blockY += 5) {
                 let pt = "(" + blockX + ", " + blockY + ")";
-                CTX.fillText(pt, gameX - CAMERA.x, gameY - CAMERA.y, Block.SCALED_SIZE);
+                CTX.fillText(pt, gameX - CAMERA.pos.x, gameY - CAMERA.pos.y, Block.SCALED_SIZE);
```
Guessing this is also for the grid @B1GD3VD0G ? To display the coordinates of the blocks?

calebkrauter commented on js/classes/vector.js:
```diff
@@ -0,0 +1,153 @@
+     * @returns {number} the Vector's magnitude
+     */
+    static magnitude(v) {
+        return Math.sqrt(v.x ** 2, v.y ** 2);
```
Didn't know that ** was an exponentiation operator! Cool.

calebkrauter commented on js/classes/vector.js:
```diff
@@ -0,0 +1,153 @@
+     * @param {Vector} b the second Vector
+     * @returns {number} the distance between a and b
+     */
+    static distance(a, b) {
```
Should this be the absolute value of the subtraction between a and b @tclaar ?

calebkrauter reviewed (commented):
Good work @tclaar ! Some minor adjustments and considerations could be made to naming. Overall this is really good! I just have some questions for clarification.
